app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt

```diff
@@ -26,44 +33,100 @@ class SellLockedFragment :
     SellLockedContract.View {
 
     companion object {
-        fun create(arguments: SellLockedArguments): SellLockedFragment =
-            SellLockedFragment()
-                .withArgs(ARG_SELL_LOCKED to arguments)
+        fun create(details: SellTransactionDetails): SellLockedFragment {
+            require(details.status == SellTransactionStatus.WAITING_FOR_DEPOSIT) {
+                "This fragment is used only if status == waiting for deposit"
+            }
+            return SellLockedFragment()
+                .withArgs(ARG_SELL_LOCKED to details)
```

Comment:
Weird.
Maybe you just create a function only for the determined status then?

Reply:
it's not a sealed class, i can't pass only determined status there 
i will cast it to sealed class later, in another task, there will be a new field there

Reply:
Yeah, would be nice. 
We should avoid throwing exception when creating a fragment